### ISSUE TITLE
adds the prefetch attribute on internal Link

### ISSUE THREAD

**Issue by simbas** [JulianCataldo/web-garden#75]
Hello @JulianCataldo,

Thank you for your awesome astro components. I was looking for a Link component which will adds the `rel=nofollow` attribute on anchors targeting external pages but also `rel=prefetch` for internal pages, for prefetching with [@astrojs/prefetch](https://github.com/withastro/astro/tree/main/packages/integrations/prefetch/). Would you be interested in this change ?

**Comment by JulianCataldo** [comment]
Interesting idea! Thank you.

Indeed the solution I've put up is much more basic than the official integration
https://github.com/JulianCataldo/web-garden/tree/develop/app/HoverPrefetch

I think it didn't existed at that time, or they were released at roughly the same time, as it was a recurring demand from users (hence, why I did it :p).

Using the `rel=prefetch` makes the process fine-grained, I guess.
I admit I didn't chimed in much in that topic as I found myself not using prefetching much.
I work mainly with SSR now (where it's not a good fit), but I could get back to this topic in a indeterminate amount of time, though :p.

I certainly would welcome a PR with pleasure, with a little line in the docs explaining the feature / behavior ^^
